mypy/subtypes.py

```diff
@@ -463,10 +463,15 @@ def visit_instance(self, left: Instance) -> bool:
                         assert unpacked.type.fullname == "builtins.tuple"
                         if isinstance(get_proper_type(unpacked.args[0]), AnyType):
                             return not self.proper_subtype
-                if mapped.type.fullname == "builtins.tuple" and isinstance(
-                    get_proper_type(mapped.args[0]), AnyType
-                ):
+                if all(isinstance(get_proper_type(a), AnyType) for a in mapped.args):
                     return not self.proper_subtype
+                if mapped.type.tuple_type:
```

Comment:
This additional branch looks like a hack. We shouldn't need this. I think the real bug is that we are passing incomplete information when accessing `__call__` on a tuple type. This diff I think is a (much) more cleaner fix:
```diff
--- a/mypy/checkexpr.py
+++ b/mypy/checkexpr.py
@@ -1476,6 +1476,7 @@ class ExpressionChecker(ExpressionVisitor[Type]):
         callable_node: Expression | None = None,
         callable_name: str | None = None,
         object_type: Type | None = None,
+        original_type: Type | None = None,
     ) -> tuple[Type, Type]:
         """Type check a call.
 
@@ -1538,7 +1539,7 @@ class ExpressionChecker(ExpressionVisitor[Type]):
                 is_super=False,
                 is_operator=True,
                 msg=self.msg,
-                original_type=callee,
+                original_type=original_type or callee,
                 chk=self.chk,
                 in_literal_context=self.is_literal_context(),
             )
@@ -1579,6 +1580,7 @@ class ExpressionChecker(ExpressionVisitor[Type]):
                 callable_node,
                 callable_name,
                 object_type,
+                original_type=callee,
             )
         else:
             return self.msg.not_callable(callee, context), AnyType(TypeOfAny.from_error)
```